pkgs/development/python-modules/IPy/default.nix

```
@@ -1,12 +1,14 @@
-{ lib, buildPythonPackage, fetchPypi, nose }:
+{ lib, buildPythonPackage, fetchFromGitHub, nose }:
 
 buildPythonPackage rec {
   pname = "IPy";
   version = "1.01";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "edeca741dea2d54aca568fa23740288c3fe86c0f3ea700344571e9ef14a7cc1a";
+  src = fetchFromGitHub {
+     owner = "autocracy";
+     repo = "python-ipy";
+     rev = "IPy-1.01";
```

I don't maintain Python stuff in Nixpkgs so cannot properly review this PR.

But... maybe the following way is more preferred? (and same for other packages)

```suggestion
     rev = "IPy-${version}";
```

---

You are absolutely right. It will be harder to do it automatically, though.